contracts/src/factory/HyperdriveFactory.sol

```diff
@@ -871,14 +883,20 @@ contract HyperdriveFactory is IHyperdriveFactory {
             revert IHyperdriveFactory.InvalidPositionDuration();
         }
 
-        // Ensure that the specified fees are within the minimum and maximum fees.
+        // Ensure that the specified fees are within the minimum and maximum
+        // fees. The flat fee is annualized so that it is consistent across all
+        // term lengths.
         if (
             _config.fees.curve > _maxFees.curve ||
-            _config.fees.flat > _maxFees.flat ||
+            // NOTE: Round down to make the check stricter.
+            _config.fees.flat.mulDivDown(365 days, _config.positionDuration) >
```

Comment:
Let's say the position duration is `365 days / 2`. If we wanted an annualized flat fee of 5 bps, we'd use `0.00025e18` as the flat fee parameter. When we annualize it here, we will compute:

```
annualizedFlatFee = 0.00025e18.mulDivDown(365 days, 365 days / 2) = 2 * 0.00025e18 = 0.0005e18 = 5 bps
```

Reply:
okay so we have to use 2.5bps input for a 5bps annualized fee if the posish duration is 6 months? would it be easier to just like always enter in the annualized one and scale it according to the position duration?

Reply:
I thought about that. The thing I didn’t love about it is that the config you pass in isn’t the config you get out, so it’s just as much of a footgun as this IMO. That said, I do see the appeal, so I can do this if you feel strongly about it. Something that might be the best of both worlds is passing the flatFeeAPR as a parameter, checking that the flat fee is 0 in the config, and overriding it with the calculated flat fee (similar to what we do with time stretch).

Reply:
it isn't a hill i wanna die on. this is something that we need to make note of though when people deploy

Reply:
100%